pinot-core/src/main/java/org/apache/pinot/server/access/BasicAuthAccessFactory.java

```diff
@@ -37,10 +40,16 @@
 
   private AccessControl _accessControl;
 
+  @Override
   public void init(PinotConfiguration configuration) {
-    _accessControl = new BasicAuthAccessControl(BasicAuthUtils.extractBasicAuthPrincipals(configuration, PREFIX));
+//    _accessControl = new BasicAuthAccessControl(BasicAuthUtils.extractBasicAuthPrincipals(configuration, PREFIX));
   }
 
+  public void init(ZkHelixPropertyStore<ZNRecord> propertyStore) {
+    _accessControl = new BasicAuthAccessControl(new UserCache(propertyStore));
+  }
```

Comment:
same as broker and controller. extend the interface and default to legacy behavior. also, move the implementation to its own class.

Reply:
I have extended the interface and default to legacy behavior. And move the implementation to its class. Please check it.